### DISCUSSION TITLE
keepIfEmpty not working as explained for filterDeep

### DISCUSSION THREAD

**Discussion by aariacarterweir** [YuriGor/deepdash#96] [Q&A]
Could just be how I've interpreted it, but:

`_.filterDeep(arr, Boolean, { leavesOnly: true, onFalse: { keepIfEmpty: true } })`

still removes empty objects

**Comment by YuriGor** [comment]
Hi, could you also provide sample data?
If you have initially empty object as a value it's treated as a leaf also, so maybe it's filtered out as not a boolean.

**Comment by YuriGor** [comment]
No response, closing for now.

**Comment by aariacarterweir** [comment]
Sorry for not replying sooner, I found that this did what I needed it to do:

`_.filterDeep(data, Boolean, { leavesOnly: true, onUndefined: { keepIfEmpty: true } })`

**Comment by YuriGor** [comment]
Cool, welcome back if anything
